### DISCUSSION TITLE
What can contribute to stale information in UI and to long refresh times?

### DISCUSSION THREAD

**Discussion by andrii-korotkov-verkada** [argoproj/argo-cd#18525] [Q&A] [answered]
Hello,
I hope your day is going well. 
I often encounter an issue that ArgoCD UI information is outdated comparing to info from `kubectl` and `kubectl argo rollouts`. Sometimes it was even keep switching back and forth between more stale and less stale views. Here might be one related bug https://github.com/argoproj/argo-cd/issues/18500 (config map is created in the cluster but shown as missing in the UI). What can contribute to this? Are there any cache durations I can tune here https://argo-cd.readthedocs.io/en/latest/operator-manual/argocd-cmd-params-cm-yaml/?
I also noticed that app refresh can be very slow taking tens of minutes sometimes or just getting stuck for 30m+. Maybe that's related to UI having stale info.
Thank you.

**Comment by Jay-J0shi** [comment]
@crenshaw-dev @agaudreault  If you guys have any context, also related - Does a app refresh restart if a watched resource has changed?

**Reply by agaudreault** [reply, marked as answer]
When a watched resource has changed, it will trigger a refresh. So a lot of changes to dependant resources can cause a constant refresh state in an application. You should look at https://argo-cd.readthedocs.io/en/stable/operator-manual/reconcile/ to optimize the behavior. It also contains some tips on which logs to look at to investigate. 

Also other applications that constantly refresh can cause the refresh queue to fill up, adding delays to an Application refresh.
